File: library/src/main/java/com/chad/library/adapter/base/animation/SlideInLeftAnimation.java
package com.chad.library.adapter.base.animation;

import android.animation.ObjectAnimator;
import android.view.View;


/**
 * https://github.com/CymChad/BaseRecyclerViewAdapterHelper
 */
public class SlideInLeftAnimation implements BaseAnimation {

    @Override public void startAnimator(View view) {
        ObjectAnimator.ofFloat(view, "translationX", -view.getRootView().getWidth(), 0).setDuration(DEFAULT_DURATION).start();
    }
}

File: library/src/main/java/com/chad/library/adapter/base/animation/SlideInLeftAnimation.kt
package com.chad.library.adapter.base.animation

import android.animation.Animator
import android.animation.ObjectAnimator
import android.view.View
import android.view.animation.DecelerateInterpolator

/**
 * https://github.com/CymChad/BaseRecyclerViewAdapterHelper
 */
class SlideInLeftAnimation : BaseAnimation {
    override fun animators(view: View): Array<Animator> {
        val animator = ObjectAnimator.ofFloat(view, "translationX", -view.rootView.width.toFloat(), 0f)
        animator.duration = 400L
        animator.interpolator = DecelerateInterpolator(1.8f)
        return arrayOf(animator)
    }
}